ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -453,6 +473,33 @@ class SignalSpec: QuickSpec {
 				observer.sendNext(1)
 				expect(lastValue).to(equal("2"))
 			}
+
+			it("should not keep resulting signal alive indefinitely") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal.never.map { $0 }
+
+				expect(signal).to(beNil())
+			}
+
+			it("should not keep resulting signal alive indefinitely even if it has an observer") {
```

Comment:
@NachoSoto What that means is that the signal is kept alive by its generator observer. If it has no generator observer, then it cannot terminated anymore because there is no way to have new events. In that case there are two possibilities: it either leaks or can deallocate. This PR does exactly the latter. That comment doesn't state anything about the signal being observed or that a disposable from an observation is being retained somewhere (this is what kept it from deallocation). In fact, this test case succeeds even without this PR because it does not retain the disposable. It's the next case that failed.
Here are some relevant lines from the Signal contract (https://github.com/ReactiveCocoa/ReactiveCocoa/blob/master/Documentation/DesignGuidelines.md#the-signalproducer-contract):
> The work associated with a Signal does not start or stop when observers are added or removed

> A signal is retained until the underlying observer is released

(underlying observer is generator observer, not observers of the signal)

> Note that it is possible to release a signal before a terminating event has been sent upon it. This should usually be avoided, as it can result in resource leaks, but is sometimes useful to disable termination.